### PR TITLE
Marketing survey for auto-renewal cancellation.

### PR DESCRIPTION
**Pull request by southp**
#### Changes proposed in this Pull Request

This PR implements a new marketing survey specifically for the auto-renewal cancellation:
demo-auto-renewal-survey

Currently we are reusing `<CancelPurchaseForm>`. However, it is confusing sometimes because its original purpose doesn't match.

In sum, this PR consists of:

1. A new component, `<CancelAutoRenewalForm>`, for the auto-renewal cancellation survey. The content is directly brought from what @semihakocer has proposed in p2-p9jf6J-1GZ from the very beginning.
1. The survey data uses a new ID, `cancel-auto-renewal`, instead of reusing `remove-calypso-purchase`. It ended up very hard to segment the data using the same ID.
1. The code for the auto-renewal cancellation survey has been removed from `<CancelPurchaseForm>`.

#### Testing instructions

1. Turn off auto-renewal and the new survey should appear.
1. Open the network inspector, submit the form, and see the POST request to `/marketing/survey` contains the data as expected.

### CODE REVIEW

File: client/components/marketing-survey/cancel-auto-renewal-form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	constructor( props ) {
+		super( props );
+
+		const { translate } = props;
+
+		this.radioButtons = [
+			[ 'take-a-break', translate( "Yes, I'm just taking a break for now." ) ],
+			[ 'manual-renew', translate( 'Yes, I want to renew manually.' ) ],
+			[ 'stop-renew', translate( 'No, I don’t have any plans to renew it.' ) ],
+			[ OTHER_FEEDBACK, translate( 'Another reason' ) ],
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-auto-renewal-form/index.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'Another reason:' )` *ES Score: 11*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/enhance/auto-renewal-survey)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/marketing-survey/cancel-auto-renewal-form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+	radioButtons = {};
+
+	constructor( props ) {
+		super( props );
+
+		const { translate } = props;
+
+		this.radioButtons = [
+			[ 'take-a-break', translate( "Yes, I'm just taking a break for now." ) ],
```
Comment (dzver) [inline client/components/marketing-survey/cancel-auto-renewal-form/index.jsx]: Out of curiosity, why "I'm" and "don’t" use different single quotes?

**Comment by michelleweber** [comment]
For copy review, could you either share static images or just the copy here, to make it easier to read? I'd love to make some tweaks to this (and to any other survey that uses this language). Thanks!

**Comment by southp** [comment]
Sorry about that, @michelleweber . Here is the screenshot:

image

File: client/components/marketing-survey/cancel-auto-renewal-form/index.jsx (hunk 2 above)
Comment (southp) [reply client/components/marketing-survey/cancel-auto-renewal-form/index.jsx]: Nice catch! It was simply a mistake from copying & pasting. Updated in a8fe4d0.

**Comment by southp** [comment]
@dzver Thanks for the quick review!
I missed one thing in the previous version though: There should be a "Need help? Chat with us." button. I have added it in 91ec522 and it looks like the screenshot in the above comment.

It shouldn't be a huge deal, but it would be great to have you double-check 🙇🏻

Fwiw, the easiest way to force it to appear is to hard-code both [isChatAvailable()](https://github.com/Automattic/wp-calypso/blob/master/client/state/happychat/selectors/is-happychat-available.js#L18) and [isPrecancellationChatAvailable()](https://github.com/Automattic/wp-calypso/blob/master/client/state/happychat/selectors/is-precancellation-chat-available.js#L14) selectors to always return true.

**Comment by michelleweber** [comment]
Thanks, @southp! 

A question on the copy: "Do you think you might use your subscription after it expires?" -- this is an inherently confusing question, because how can I use an expired subscription?

Is this really a way of asking "do you think you'll manually renew your subscription?" Is what we want to know here is not whether they're interested in the plan in particular, but whether turning off auto-renew is because they don't need the product or because they just don't want it to bill automatically?

**Comment by southp** [comment]
That's some great feedback, @michelleweber . From what I understanding, the latter:

> whether turning off auto-renew is because they don't need the product or because they just don't want it to bill automatically?

is what we want to know. In that case, what would you suggest us to put as the question?

**Comment by michelleweber** [comment]
A quick question...

Auto-renewal is now off. Before you go, we'd love to know: are you letting this plan expire completely, or do you think you'll renew it manually?

- I'm going to let this plan expire.
- I'm going to renew the plan, but will do it manually.
- I'm not sure.

**Comment by mktgmolly** [comment]
Yes to @michelleweber's changes that make the copy clear and friendly! @southp do you have any concerns or is there anything else we need to consider (like how/where these responses are being recorded) before changing the copy and selection options? Thank you!

**Comment by southp** [comment]
Thanks for the feedback, @michelleweber . With 8d950cf, it now looks like this:
image

Is this what you meant? I'm not sure if you meant to remove the "Another reason" option together as well, so I'm keeping it at the moment. Please feel free to let me know if you want me to remove it. I'm fine either way :)

**Comment by michelleweber** [comment]
Yes, please remove the "another reason" -- it doesn't apply here, because this is basically a yes-or-no question where we haven't asked for reasons.

**Comment by southp** [comment]
Thanks for the clarification. With c342f30, it now looks like this:
image
It should be the form we are agreeing on, so I'm going to deploy it this way. Please feel free to revise if I'm wrong :)

**Comment by niranjan-uma-shankar** [comment]
The copy needs to be different for non-plan products. Issue described here - https://github.com/Automattic/wp-calypso/issues/35224

**Pull request merged**